Enable opt out for connection caching

nmaeder opened the pull request:
Allow for opting out of the connection caching. No change in the default behavior.

Useful to prevent unused open connection when using the lwreg utility in longer python scripts without calling the private `_clear_cached_connection()`.

greglandrum reviewed (changes requested):
I think the idea makes sense, but adding it as an argument (that someone has to remember) for every function seems wrong.

Why not add this as a new entry of the `config` dict?

nmaeder commented:
good idea, will implement that accordingly.

nmaeder commented:
Moved it to the config and added an entry to the default config.
It will not be stored in the metadata, since this is usecase specific. I left the kwarg in configureFromDatabase to allow this to be closed (needed since its not in the tables metadata) or should i remove that?

greglandrum reviewed (approved):
LGTM

greglandrum commented:
@brje01 are you also :+1: on this one?

greglandrum commented on lwreg/utils.py:
```diff
@@ -251,7 +254,8 @@ def connect(config):
     else:
         _replace_placeholders = _replace_placeholders_noop
-    _dbConnection = cn
+    if _lookupWithDefault(config, "cacheConnection"):
+        _dbConnection = cn
```
The `configure_from_database()` function will open a cached connection when it is called.
It makes sense here to clear any existing cached connections if we disable caching.
```suggestion
    if _lookupWithDefault(config, "cacheConnection"):
        _dbConnection = cn
    else:
        _dbConnection = None
```

nmaeder replied:
I left the kwarg `cache_connection` in `configure_from_database()` to be able to prevent that from happening.

greglandrum reviewed (approved):
LGTM
